src/modules/i18n/components/LangButton.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+    <button
+        class="language-btn btn"
+        :class="buttonClass"
+        :title="lang.toUpperCase()"
+        :data-cy="`menu-lang-${lang}`"
+        @click="changeLang"
+    >
+        {{ lang.toUpperCase() }}
+    </button>
+</template>
+
+<script>
+import log from '@/utils/logging'
+import { mapActions, mapGetters, mapState } from 'vuex'
+
+export default {
+    props: {
+        lang: {
+            type: String,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            selected: false,
+        }
+    },
+    computed: {
+        ...mapGetters(['isDesktopMode']),
+        ...mapState({
+            currentLang: (state) => state.i18n.lang,
+        }),
+        buttonClass() {
+            let classes = ''
+            if (this.isDesktopMode) {
+                classes += 'm-0 px-1 btn-xs btn-link custom-text-decoration'
+                if (this.selected) {
+                    classes += ' text-primary'
+                } else {
+                    classes += ' text-black'
+                }
+            } else {
+                classes += ' mobile-view btn-sm mx-1'
+                if (this.selected) {
+                    classes += ' btn-primary'
+                } else {
+                    classes += ' btn-light'
+                }
+            }
+            return classes
+        },
```

Comment:
you can make an array of class if you want, or you could directly state classes in the template with 
```
<button
        class="language-btn btn"
        :class="{
            "m-0 px-1 btn-xs btn-link custom-text-decoration text-black": isDesktopMode,
            "text-primary": isDesktopMode && selected,
            "mobile-view btn-sm mx-1 btn-light": !isDesktopMode,
            "btn-primary": !isDesktopMode && selected
        }"
</button
```

If possible I tend to prefer this way of expressing classes, so that their definition stays in the HTML and is not mixed with the JS.

I'm not 100% sure about how `btn-primary` or `text-primary` will be interpreted with how I've written the class props above, maybe you will need to declare `btn-light` and `text-black` on a separate entry (they might not be overridden by the primary classes...)

Reply:
👍🏼

Reply:
Good idea I also did not like the way I did it but did not thought of this way above 😉